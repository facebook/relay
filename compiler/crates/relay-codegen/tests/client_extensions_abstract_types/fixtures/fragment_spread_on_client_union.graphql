query Foo {
  client_type {
    ...MyFragment
  }
}

fragment MyFragment on ClientUnion {
  __typename
}

# %extensions%

extend type Query {
  client_type: ClientType
}

type ClientType {
  name: String
}

type OtherClientType {
  name: String
}

union ClientUnion = ClientType | OtherClientType
