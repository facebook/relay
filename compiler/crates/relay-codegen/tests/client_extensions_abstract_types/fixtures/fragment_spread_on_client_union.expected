==================================== INPUT ====================================
query Foo {
  client_type {
    ...MyFragment
  }
}

fragment MyFragment on ClientUnion {
  __typename
}

# %extensions%

extend type Query {
  client_type: ClientType
}

type ClientType {
  name: String
}

type OtherClientType {
  name: String
}

union ClientUnion = ClientType | OtherClientType
==================================== OUTPUT ===================================
{
  "argumentDefinitions": [],
  "kind": "Fragment",
  "metadata": null,
  "name": "MyFragment",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "__typename",
      "storageKey": null
    }
  ],
  "type": "ClientUnion",
  "abstractKey": "__isClientUnion"
}

{
  "argumentDefinitions": [],
  "kind": "Operation",
  "name": "Foo",
  "selections": [
    {
      "alias": null,
      "args": null,
      "concreteType": "ClientType",
      "kind": "LinkedField",
      "name": "client_type",
      "plural": false,
      "selections": [
        {
          "args": null,
          "kind": "FragmentSpread",
          "name": "MyFragment"
        }
      ],
      "storageKey": null
    }
  ],
  "clientAbstractTypes": {
    "__isClientUnion": [
      "ClientType",
      "OtherClientType"
    ]
  }
}
